docs/rfcs/1013-parsec-account.md

```diff
@@ -26,485 +36,758 @@ That is how we came to the idea of storing the device on a remote server.
+> [!IMPORTANT]
+> We use the URL-safe `base64` variant.
 
-On `ok`, the server would have sent a mail with a unique token used for next the request used to register the authentication method.
+The signature is generated like so:
+
+```math
+\begin{gather}
+body\_sha256 = sha256(body) \\
+content = \text{"PARSEC-PASSWORD-MAC-BLAKE2B"} \Vert \text{"."} \Vert base64(email) \Vert \text{"."} \Vert timestamp \Vert \text{"."} \Vert body\_sha256 \\
+code = hmac_{black2b}(shared\_secret, content) \\
+signature = base64(code)
+\end{gather}
+```
 
-> The token will be valid for a duration that will be defined at the implementation (likely will be configurable)
+> [!TIP]
+> $\Vert$ is the concatenation operator.
 
-If the email is already registered in the service, a `ok` response will still be sent (without sending an email)
-to avoid creating an oracle about emails registered in the service.
+> Note: We pass the checksum of the body and not the body itself to simplify generating
+> the signature on low-power device (smartcard) since they would have less data to process.
+
+!!! TODO !!!
+- Remove sha256 ?
+- Remove checksum of the body ?
+
+### 4.3 - Account creation: validate email
+
+To create an account, the client starts by requesting an email validation token.
 
-```yml
+Anonymous account API:
+
+```json5
 {
-  "cmd": "account_password_create",
+  "cmd": "account_create_send_validation_email",
   "req": {
-    "email_validation_token": "EmailValidationToken",
-    "account_label": "String",
-    "auth_method_key_params": "KeyAlgorithm",
-    "auth_method_hmac_key": "HMACKey",
-    "encrypted_vault_key": "EncryptedSymKey"
+    "fields": [
+      {
+        "name": "email",
+        "type": "EmailAddress"
+      }
+    ]
   },
   "reps": [
     {
-      "status": "ok",
+      "status": "ok"
     },
     {
-      "status": "invalid_email_validation_token"
+      // The email could not be sent due to SMTP server issue
+      "status": "email_server_unavailable"
+    },
+    {
+      // The SMTP server rejected the email recipient
+      "status": "email_recipient_refused"
     }
   ]
 }
 ```
 
-> This request is authenticated with the newly generated key.
-> But it's possible that we make it unauthenticated in the implementation to not have a dedicated verification process for the request
-> (since it would need to use the data provided in the request vs save in the server).
+> Note: `EmailAddress` is a new type allowing to validate the email address format.
 
-Upon receiving the request, the server will create the [`Account`], [`AuthMethod`], and [`Vault`].
+On `ok`, the server would have sent a mail with a unique token used for next the request used to register the authentication method.
 
-The IP address and User-agent of the creator are recorded (see Data Model).
+If an account already exists with this email, a `ok` response will still be sent (without sending an email)
+to avoid creating an oracle about emails registered in the service.
```

Comment:
Should we add precision on how we avoid re sending an email too soon after the initial request was made ?

```
If an account creation has already been requested for an email and the email is not validated yet, a new email with a new token may or may not been sent depending on how long ago the previous one was sent.
```